src/blmc_joint_module.cpp

```diff
@@ -298,6 +298,25 @@ bool BlmcJointModule::calibrate(double& angle_zero_to_index,
     return true;
 }
 
+void BlmcJointModule::homing_at_current_position(int joint_id,
+                                                 double home_offset_rad)
+{
+    // reset the internal zero angle.
+    set_zero_angle(0.0);
+
+    // TODO: would be nice if the joint instance had a `name` or `id` and class
+    // level instead of storing it here (to make more useful debug prints).
+    homing_state_.joint_id = joint_id;
+
+    homing_state_.home_offset_rad = home_offset_rad;
+    homing_state_.target_position_rad = get_measured_angle();
+
+    // set the zero angle
+    set_zero_angle(homing_state_.target_position_rad - homing_state_.home_offset_rad);
+
+    homing_state_.status = HomingReturnCode::SUCCEEDED;
```

Comment:
I think the usage of `homing_state_` could completely be avoided here.  As this homing method is finished with a single call, there is no need to keep any state information for later.

Reply:
Of course